cvss40.js

```diff
@@ -0,0 +1,1752 @@
+            const AC = this.getEffectiveMetricValue("AC");
+            const AT = this.getEffectiveMetricValue("AT");
+
+            return (AC === "L" && AT === "N") ? "0" : "1";
+        };
+
+        // Helper function to compute EQ3
+        const computeEQ3 = () => {
+            const VC = this.getEffectiveMetricValue("VC");
+            const VI = this.getEffectiveMetricValue("VI");
+            const VA = this.getEffectiveMetricValue("VA");
+
+            if (VC === "H" && VI === "H") {
+                return "0";
+            }
+            if (!(VC === "H" && VI === "H") && (VC === "H" || VI === "H" || VA === "H")) {
+                return "1";
+            }
+            if (!(VC === "H" || VI === "H" || VA === "H")) {
+                return "2";
+            }
+        };
+
+        // Helper function to compute EQ4
+        const computeEQ4 = () => {
+            const MSI = this.getEffectiveMetricValue("MSI");
+            const MSA = this.getEffectiveMetricValue("MSA");
+            const SC = this.getEffectiveMetricValue("SC");
+            const SI = this.getEffectiveMetricValue("SI");
+            const SA = this.getEffectiveMetricValue("SA");
+
+            if (MSI === "S" || MSA === "S") {
+                return "0";
+            }
+            if (!(MSI === "S" || MSA === "S") && (SC === "H" || SI === "H" || SA === "H")) {
+                return "1";
+            }
+            return "2";
+        };
+
+        // Helper function to compute EQ5
+        const computeEQ5 = () => {
+            const E = this.getEffectiveMetricValue("E");
+            if (E === "A") return "0";
+            if (E === "P") return "1";
+            if (E === "U") return "2";
+        };
+
+        // Helper function to compute EQ6
+        const computeEQ6 = () => {
+            const CR = this.getEffectiveMetricValue("CR");
+            const VC = this.getEffectiveMetricValue("VC");
+            const IR = this.getEffectiveMetricValue("IR");
+            const VI = this.getEffectiveMetricValue("VI");
+            const AR = this.getEffectiveMetricValue("AR");
+            const VA = this.getEffectiveMetricValue("VA");
+
+            if ((CR === "H" && VC === "H") || (IR === "H" && VI === "H") || (AR === "H" && VA === "H")) {
+                return "0";
+            }
+            return "1";
+        };
+
+        // Compute all equivalency values
+        const eq1 = computeEQ1();
+        const eq2 = computeEQ2();
+        const eq3 = computeEQ3();
+        const eq4 = computeEQ4();
+        const eq5 = computeEQ5();
+        const eq6 = computeEQ6();
+
+        // Combine all EQ values into the equivalency vector
+        return eq1 + eq2 + eq3 + eq4 + eq5 + eq6;
+    }
+}
+
+// Check if the environment is Node.js
+if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') {
+    module.exports = { CVSS40 };
+} else {
+    // In a browser environment, attach CVSS40 to the window object
+    window.CVSS40 = CVSS40;
+}
```

Comment:
This is a nice touch. We would [benefit from adding a new line/LF at the end of the file, however.](https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log)

Reply:
I will certainly add the newline/LF at the end of the file as suggested.